Source/SwiftLintFramework/Rules/Style/ExplicitReturnRule.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import SourceKittenFramework
+import SwiftSyntax
+
+public struct ExplicitReturnRule: ConfigurationProviderRule, SubstitutionCorrectableRule, OptInRule {
+    public var configuration = ExplicitReturnConfiguration()
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_return",
+        name: "Explicit Return",
+        description: "Prefer explicit returns in closures, functions and getters.",
+        kind: .style,
+        nonTriggeringExamples: ExplicitReturnRuleExamples.nonTriggeringExamples,
+        triggeringExamples: ExplicitReturnRuleExamples.triggeringExamples,
+        corrections: ExplicitReturnRuleExamples.corrections
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        return violationOffsets(file: file).map {
+            StyleViolation(
+                ruleDescription: Self.description,
+                severity: configuration.severityConfiguration.severity,
+                location: Location(file: file, byteOffset: $0)
```

Comment:
it'd be good to customize the `reason` based on `returnKind` so messages are more accurate like "Prefer explicit returns in functions" for a function

Reply:
That's a brilliant idea! I did this that way to be consistent with the opposite, upstreamed rule: `implicit_return`. I believe if we want to add such differentiation here, we should also do it for the `implicit` rule. I'd rather do this separately, although if you think it should be done in this PR I can still do this.